### PR TITLE
fix(provider): wrong container image name

### PR DESCRIPTION
**Pull request by paralta**
## Description

The Container Image name is available inside the `RepoTag` field. Included in this PR:
* Make container and container image id a required field in the API spec
* Remove prefix slash from container name
* Replace image name with repotags and repodigest fields
* Add container info and container image info to UI asset type items
* Extend basic test to scan a docker image

Related to https://github.com/openclarity/vmclarity/issues/763, but does not fix UI issue.

## Type of Change

[x] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Comment by paralta** [comment]
Currently looking into the new test failure 👀

File: api/openapi.yaml [hunk 1]
```diff
@@ -2214,6 +2214,7 @@ components:
           nullable: true
       required:
         - objectType
+        - id
```
Comment (fishkerez) [inline api/openapi.yaml:2217]: What does "id" represent here? it is not the database id, right?

File: e2e/basic_scan_test.go [hunk 2]
```diff
@@ -45,66 +44,29 @@ var _ = ginkgo.Describe("Running a basic scan (only SBOM)", func() {
...
+	ginkgo.Context("which scans a docker image", func() {
+		ginkgo.It("should finish successfully", func(ctx ginkgo.SpecContext) {
+			ginkgo.By("waiting until test asset is found")
+			filter := "assetInfo/repoTags/any(s:startswith(s, 'alpine:3.18.2'))"
 			reportFailedConfig.objects = append(
 				reportFailedConfig.objects,
-				APIObject{"scanConfig", fmt.Sprintf("id eq '%s'", *apiScanConfig.Id)},
+				APIObject{"asset", filter},
 			)
-
```
Comment (fishkerez) [inline e2e/basic_scan_test.go:66]: Are these changes related to the PR?

File: api/openapi.yaml (hunk 1 above)
Comment (fishkerez) [reply api/openapi.yaml:2217]: I think we should call it imageID, to differentiate from the id fields we have on all our resources which represents the database id

Comment (fishkerez) [reply api/openapi.yaml:2217]: containerID in that case

File: api/openapi.yaml [hunk 3]
```diff
@@ -2236,7 +2243,8 @@ components:
         size:
           type: integer
       required:
-        - objectType 
+        - objectType
+        - id
```
Comment (fishkerez) [inline api/openapi.yaml:2247]: Let's change the name to imageID

File: api/openapi.yaml (hunk 1 above)
Comment (paralta) [reply api/openapi.yaml:2217]: Thanks for reviewing, @fishkerez ! Agree with you here, it will also make it more consistent with `VMInfo` where we have a `instanceID` 👍

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

File: api/models/containerimageinfo.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+// limitations under the License.
+
+package models
+
+// GetRepoTag returns the first repo tag if it exists. Otherwise, returns false.
+func (c *ContainerImageInfo) GetRepoTag() (string, bool) {
+	var tag string
+	var ok bool
+
+	if len(*c.RepoTags) > 0 {
```
Comment (fishkerez) [inline api/models/containerimageinfo.go:23]: need to check for c == nil && c.c.RepoTags == nil ?

File: api/models/containerimageinfo.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+	return tag, ok
+}
+
+// GetRepoDigest returns the first repo digest if it exists. Otherwise, returns false.
+func (c *ContainerImageInfo) GetRepoDigest() (string, bool) {
+	var digest string
+	var ok bool
+
+	if len(*c.RepoDigests) > 0 {
```
Comment (fishkerez) [inline api/models/containerimageinfo.go:35]: need to check for c == nil && c.RepoDigests == nil ?

File: api/models/containerimageinfo.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	var ok bool
+
+	if len(*c.RepoTags) > 0 {
+		tag, ok = (*c.RepoTags)[0], true
+	}
+
+	return tag, ok
+}
+
+// GetRepoDigest returns the first repo digest if it exists. Otherwise, returns false.
```
Comment (fishkerez) [inline api/models/containerimageinfo.go:30]: why not returning all the repo digests and let the caller decide if he wants all or just the first one?

File: api/models/containerimageinfo.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package models
+
+// GetRepoTag returns the first repo tag if it exists. Otherwise, returns false.
+func (c *ContainerImageInfo) GetRepoTag() (string, bool) {
```
Comment (fishkerez) [inline api/models/containerimageinfo.go:19]: why not returning all the repo tags and let the caller decide if he wants all or just the first one?

File: api/models/containerimageinfo.go (hunk 4 above)
Comment (chrisgacsal) [reply api/models/containerimageinfo.go:23]: @fishkerez We can check if `c` is `nil` to make sure the receiver is not `nil`, however `len()` returns `0` if `c.RepoTags` is `nil` so the 2nd check is not needed as `len` does that implicitly.

File: api/models/containerimageinfo.go (hunk 5 above)
Comment (chrisgacsal) [reply api/models/containerimageinfo.go:35]: @fishkerez Same as above.

File: api/models/containerimageinfo.go (hunk 6 above)
Comment (chrisgacsal) [reply api/models/containerimageinfo.go:30]: This is just a helper to get the first `RepoDigest` in the list. If the caller want to get all then it can use the `c.RepoDigests`. But we can add another helper which returns all of the `RepoTags` and `RepoDigests` if there are any.

File: api/models/containerimageinfo.go (hunk 7 above)
Comment (chrisgacsal) [reply api/models/containerimageinfo.go:19]: @fishkerez Same as above.

File: api/models/containerimageinfo.go (hunk 4 above)
Comment (fishkerez) [reply api/models/containerimageinfo.go:23]: yeah, but we are dereferencing c.RepoTags (by doing *c.RepoTags), so if c.RepoTags are nil, we will crush

Comment (chrisgacsal) [reply api/models/containerimageinfo.go:23]: Testing the receiver for `nil` is not something we do in other parts of the code base as the assumption is that the caller won't call any method on `nil` pointer. Or at least it makes sure that safeguards are in place to avoid this issue. Personally I do not mind to test the receiver as well, but that we need to make sure that we follow the same principle in other parts of the code base as well.

The problem is that what should be the return value if the the receiver is null? An error or `0` value?

Comment (chrisgacsal) [reply api/models/containerimageinfo.go:23]: > yeah, but we are dereferencing c.RepoTags (by doing *c.RepoTags), so if c.RepoTags are nil, we will crush

That is valid point. @paralta Could you please fix this?

File: api/models/containerimageinfo.go (hunk 6 above)
Comment (fishkerez) [reply api/models/containerimageinfo.go:30]: Ahh ok now I see. So maybe better calling it GetFirstRepoDigest/Tag ?

File: pkg/orchestrator/provider/docker/container.go [hunk 8]
```diff
@@ -103,15 +104,17 @@ func (c *Client) getContainerInfo(ctx context.Context, containerID, imageID stri
 	}
 
 	// Get container image info
-	imageInfo, err := c.getContainerImageInfo(ctx, imageID)
+	imageInfo, err := c.getContainerImageInfo(ctx, info.Image)
 	if err != nil {
 		return models.ContainerInfo{}, err
 	}
 
+	containerName := strings.Trim(info.Name, "/")
```
Comment (fishkerez) [inline pkg/orchestrator/provider/docker/container.go:112]: Do you want Trim or TrimSuffix here?

Comment (paralta) [reply pkg/orchestrator/provider/docker/container.go:112]: It would be `TrimPrefix` in this case, but I think we are fine since docker does not allow container names with slashes

**Review by Tehsmash — CHANGES_REQUESTED** [review]
I agree with Erez's points, lets get those fixed otherwise LGTM.

File: pkg/uibackend/rest/dashboard_riskiest_assets_test.go [hunk 9]
```diff
@@ -321,7 +321,7 @@ func createContainerImageInfo(t *testing.T) *backendmodels.AssetType {
 	t.Helper()
 	info := backendmodels.AssetType{}
 	err := info.FromContainerImageInfo(backendmodels.ContainerImageInfo{
-		Name: utils.PointerTo("ghcr.io/openclarity/vmclarity-orchestrator"),
+		RepoTags: utils.PointerTo([]string{"ghcr.io/openclarity/vmclarity-orchestrator"}),
```
Comment (Tehsmash) [inline pkg/uibackend/rest/dashboard_riskiest_assets_test.go:324]: missing the tag part here

File: pkg/uibackend/rest/dashboard_riskiest_assets.go [hunk 10]
```diff
@@ -231,8 +231,15 @@ func containerInfoToAssetInfo(info backendmodels.ContainerInfo) (*models.AssetIn
...
-	repo, name := path.Split(*info.Name)
-	repo = strings.TrimSuffix(repo, "/")
+	var repo, name string
+
+	tag, ok := info.GetRepoTag()
+	if ok {
+		repo, name = path.Split(tag)
+	} else {
+		name = info.ImageID
+	}
```
Comment (Tehsmash) [inline pkg/uibackend/rest/dashboard_riskiest_assets.go:241]: In the kubernetes work I handled this slightly differently: Name is always the ID, Location is the first repo digest (if there is one). The reason for using the whole digest as the location is because the repo is like the street name, but the digest is like the house number. Its an address where you can find the image with this ID if you curl it.

Comment (chrisgacsal) [reply pkg/uibackend/rest/dashboard_riskiest_assets.go:241]: Lets have the same logic as @Tehsmash discribed above.

**Review by Tehsmash — APPROVED** [review]
